net/netcheck/netcheck.go

```diff
@@ -1110,22 +1109,25 @@ func (c *Client) runHTTPOnlyChecks(ctx context.Context, last *Report, rs *report
 	return nil
 }
 
+// measureHTTPSLatency measures the initial TLS connection latency to the DERP
+// region, but only returns success if an HTTPS request to the region succeeds.
 func (c *Client) measureHTTPSLatency(ctx context.Context, reg *tailcfg.DERPRegion) (time.Duration, netip.Addr, error) {
 	metricHTTPSend.Add(1)
-	var result httpstat.Result
-	ctx, cancel := context.WithTimeout(httpstat.WithHTTPStat(ctx, &result), httpsProbeTimeout)
+	ctx, cancel := context.WithTimeout(ctx, httpsProbeTimeout)
 	defer cancel()
 
 	var ip netip.Addr
 
 	dc := derphttp.NewNetcheckClient(c.logf, c.NetMon)
 	defer dc.Close()
 
+	startTime := c.timeNow()
 	tlsConn, tcpConn, node, err := dc.DialRegionTLS(ctx, reg)
```

Comment:
This dials nodes sequentially (see `dialRegion()`. If a node is unavailable we can be subjected to timeouts inflating the TCP+TLS dial time, which can push us away from the preferred region.

Reply:
good point. As that's a larger fix, I'd like to fix it in a secondary issue, focusing here on fixing the data race. Sound reasonable? I'll file a bug for it

Reply:
SGTM